### PR TITLE
Add a log level check simply before logging.

### PR DESCRIPTION
**Pull request by seedeed**
Motivation:

ThrowableUtil.stackTraceToString is an expensive method call. So I think a log level check before this logging statement is quite needed especially in a environment with the warning log disabled.

Modification:

Add log level check simply before logging.

Result:

Improve performance in a environment with the warning log disabled.

### CODE REVIEW

File: common/src/main/java/io/netty/util/internal/PromiseNotificationUtil.java [hunk 1]
```diff
@@ -66,9 +66,11 @@ public static void tryFailure(Promise<?> p, Throwable cause, InternalLogger logg
             if (err == null) {
                 logger.warn("Failed to mark a promise as failure because it has succeeded already: {}", p, cause);
             } else {
-                logger.warn(
+                if (logger.isWarnEnabled()) {
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/PromiseNotificationUtil.java]: Just merge this with the else and make it an else if?

Comment (seedeed) [reply common/src/main/java/io/netty/util/internal/PromiseNotificationUtil.java]: Got it

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@seedeed thanks a lot

**Pull request merged**